compiler/src/dotty/tools/dotc/transform/TreeChecker.scala

```diff
@@ -657,8 +657,8 @@ object TreeChecker {
 
       // Check that we only add the captured type `T` instead of a more complex type like `List[T]`.
       // If we have `F[T]` with captured `F` and `T`, we should list `F` and `T` separately in the args.
-      for arg <- args do
-        assert(arg.isTerm || arg.tpe.isInstanceOf[TypeRef], "Expected TypeRef in Hole type args but got: " + arg.tpe)
+      for arg <- args if arg.isType do
+        assert(arg.tpe.isInstanceOf[TypeRef] || arg.tpe.isInstanceOf[TermRef], "Expected TypeRef or TermRef in Hole type args but got: " + arg.tpe)
```

Comment:
I don't understand this change, can you explain why we do this?

Reply:
Before, we assumed that we could only have type references, a reference to a type on some arbitrary path that is represented with a `TypeRef` (`T`, `x.T`, `y.x.T`). But we missed singleton types represented with a `TermRef` (`a.type`, `a.b.type`, ...).